packages/dashboard/src/app/views/myStories/filters/taxonomy/useTaxonomyFilters.js

```diff
@@ -0,0 +1,151 @@
+/*
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import {
+  useCallback,
+  useEffect,
+  useState,
+  useMemo,
+} from '@googleforcreators/react';
+
+/**
+ * Internal dependencies
+ */
+import useApi from '../../../../api/useApi';
+
+export const getFilterPlaceholder = (filter) => {
+  return filter.labels?.allItems || filter.labels?.name || filter.restBase;
+};
+
+export const getFilterAriaLabel = (filter) => {
+  const ariaLable =
+    filter.labels?.singularName || filter.labels?.name || filter.restBase;
+  return `Filter stories by ${ariaLable}`;
+};
```

Comment:
Note that all strings need to use `__()` so they can be translated ⚠️  Here with `sprintf` and a `translators:` comment for the placeholder.

However, here we don't need a custom label & function for it. We can just use the existing `filter.labels.filterByItem` label.

See my below suggestions that make these functions here obsolete.

---

Aside: Typo `ariaLable` -> `ariaLabel`

Reply:
I totally agree, I originally was trying to use `__()` and `sprintf` - https://github.com/GoogleForCreators/web-stories-wp/pull/11625/commits/5e63aef8a1855b804f4caa052b8b4a94515674f8#:~:text=export%20const%20getFilterPlaceholder,.key%3B - but the `i18n` linter does not allow single placeholders - https://github.com/WordPress/gutenberg/blob/trunk/packages/eslint-plugin/docs/rules/i18n-no-placeholders-only.md - and we're getting the exact wording from the backend that we want to use here.  Should we ignore the linting warning or is there a better way to handle this?

Reply:
You would need to use `__()` only when more text is involved. If it's just a variable and nothing else, you don't need to wrap it in `__()`. That's what the lint rule guards against.

If you have text like "Do something with %s" -> wrap it in `__()`

If you only have a variable like `fooBar` -> just use as-is